packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/import/vsphere-user-import.controller.js

```diff
@@ -0,0 +1,108 @@
+import { DEFAULT_FILTER_COLUMN } from './vsphere-user-import.constant';
+
+export default class {
+  /* @ngInject */
+  constructor($q, $translate, DedicatedCloud) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.DedicatedCloud = DedicatedCloud;
+  }
+
+  $onInit() {
+    this.task = null;
+
+    this.loaders = {
+      init: true,
+      action: false,
+      poll: false,
+      cancelModal: false,
+    };
+
+    this.importUser = {
+      type: null,
+      userExemple: { user: null, group: null },
+      groupName: null,
+      userName: null,
+      domainName: null,
+      activeDirectoryId: null,
+    };
+
+    const params = {
+      offset: null,
+      pageSize: null,
+      sort: DEFAULT_FILTER_COLUMN,
+      sortOrder: 'ASC',
+      filters: [],
+      defaultFilterColumn: DEFAULT_FILTER_COLUMN,
+    };
+
+    return this.$q
+      .resolve(this.DedicatedCloud.getActiveDirectories(this.productId, params))
```

Comment:
Why do we wrap it into a promise here ? Can't we just 
```suggestion
    return this.DedicatedCloud.getActiveDirectories(this.productId, params)
```

Reply:
It's applied